File: morphac/simulation/playground/python/test/playground_spec_test.py
import pytest

from morphac.simulation.playground import PlaygroundSpec


@pytest.fixture()
def generate_playground_spec_list():
    ps1 = PlaygroundSpec("playground1", 0.001)
    ps2 = PlaygroundSpec(name="playground2", dt=0.02)

    return ps1, ps2


def test_attributes(generate_playground_spec_list):
    ps1, ps2 = generate_playground_spec_list

    # Test the attributes of the Spec objects.
    assert ps1.name == "playground1"
    assert ps1.dt == 0.001

    assert ps2.name == "playground2"
    assert ps2.dt == 0.02

File: morphac/simulation/playground/test/playground_spec_test.cc
#include "simulation/playground/include/playground_spec.h"

#include "gtest/gtest.h"

namespace {

using morphac::simulation::playground::PlaygroundSpec;

class PlaygroundSpecTest : public ::testing::Test {
 protected:
  PlaygroundSpecTest() {}

  void SetUp() override {}

  PlaygroundSpec playground_spec1_{"playground1", 0.1};
  PlaygroundSpec playground_spec2_{"playground2", 0.05};
};

TEST_F(PlaygroundSpecTest, Members) {
  ASSERT_EQ(playground_spec1_.name, "playground1");
  ASSERT_EQ(playground_spec1_.dt, 0.1);

  ASSERT_EQ(playground_spec2_.name, "playground2");
  ASSERT_EQ(playground_spec2_.dt, 0.05);
}

}  // namespace

int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
